number_test.go

```diff
@@ -584,3 +590,177 @@ func TestNumber_ConvertLesser(t *testing.T) {
 	value.chain.assertFailed(t)
 	value.chain.clearFailed()
 }
+
+func TestNumber_IsInt(t *testing.T) {
+	reporter := newMockReporter(t)
+
+	t.Run("check parameters", func(t *testing.T) {
+		NewNumber(reporter, 1234).IsInt(0, 1, 2, 3).
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 1234).IsInt().
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, 1234.0001).NotInt(0, 1, 2, 3).
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 1234.0001).NotInt().
+			chain.assertNotFailed(t)
+	})
+
+	t.Run("without bit size", func(t *testing.T) {
+		NewNumber(reporter, -1234).IsInt().
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, 1234.00001).IsInt().
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 1234).NotInt().
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 1234.00001).NotInt().
+			chain.assertNotFailed(t)
+	})
+
+	t.Run("with bit size", func(t *testing.T) {
+		NewNumber(reporter, -math.MaxInt8).IsInt(32).
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, math.MaxInt64).IsInt(32).
+			chain.assertFailed(t)
+
+		NewNumber(reporter, math.MaxInt64).NotInt(32).
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, math.MaxInt8).NotInt(32).
+			chain.assertFailed(t)
+	})
+
+	t.Run("check edge cases", func(t *testing.T) {
+		NewNumber(reporter, 20).IsInt(3).
+			chain.assertFailed(t)
+
+		NewNumber(reporter, math.Inf(1)).IsInt().
+			chain.assertFailed(t)
+
+		NewNumber(reporter, math.NaN()).IsInt().
+			chain.assertFailed(t)
+
+		NewNumber(reporter, 20).NotInt(3).
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, math.Inf(-1)).NotInt().
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, math.NaN()).NotInt().
+			chain.assertNotFailed(t)
+	})
+
+	t.Run("big integer cases", func(t *testing.T) {
+		t.Skip("skipped until migration to big Float is done")
+
+		NewNumber(reporter, math.MaxInt64).IsInt().
+			chain.assertNotFailed(t)
+
+		NewNumber(reporter, math.MaxInt64).NotInt(64).
+			chain.assertFailed(t)
+	})
```

Comment:
We can remove tests for 64 bit ints at all.

Tests for lower bitnesses (32, 16, etc) will fully cover functionality of IsInt, IsUnt, etc, because they are not really tied to specific bitness.

After migration to big.Float we will revise all tests and decide where we need to cover more cases.

Reply:
Similar for IsUnit test.